// plugin-welcome/src/api/models/EntBillEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntCustomer,
    EntCustomerFromJSON,
    EntCustomerFromJSONTyped,
    EntCustomerToJSON,
    EntProduct,
    EntProductFromJSON,
    EntProductFromJSONTyped,
    EntProductToJSON,
    EntUser,
    EntUserFromJSON,
    EntUserFromJSONTyped,
    EntUserToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntBillEdges
 */
export interface EntBillEdges {
    /**
     * 
     * @type {EntCustomer}
     * @memberof EntBillEdges
     */
    customer?: EntCustomer;
    /**
     * 
     * @type {EntProduct}
     * @memberof EntBillEdges
     */
    product?: EntProduct;
    /**
     * 
     * @type {EntUser}
     * @memberof EntBillEdges
     */
    user?: EntUser;
}

export function EntBillEdgesFromJSON(json: any): EntBillEdges {
    return EntBillEdgesFromJSONTyped(json, false);
}

export function EntBillEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntBillEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'customer': !exists(json, 'Customer') ? undefined : EntCustomerFromJSON(json['Customer']),
        'product': !exists(json, 'Product') ? undefined : EntProductFromJSON(json['Product']),
        'user': !exists(json, 'User') ? undefined : EntUserFromJSON(json['User']),
    };
}

export function EntBillEdgesToJSON(value?: EntBillEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'customer': EntCustomerToJSON(value.customer),
        'product': EntProductToJSON(value.product),
        'user': EntUserToJSON(value.user),
    };
}
