// plugin-welcome/src/api/models/EntCustomerEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBill,
    EntBillFromJSON,
    EntBillFromJSONTyped,
    EntBillToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntCustomerEdges
 */
export interface EntCustomerEdges {
    /**
     * Buys holds the value of the buys edge.
     * @type {Array<EntBill>}
     * @memberof EntCustomerEdges
     */
    buys?: Array<EntBill>;
}

export function EntCustomerEdgesFromJSON(json: any): EntCustomerEdges {
    return EntCustomerEdgesFromJSONTyped(json, false);
}

export function EntCustomerEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntCustomerEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'buys': !exists(json, 'buys') ? undefined : ((json['buys'] as Array<any>).map(EntBillFromJSON)),
    };
}

export function EntCustomerEdgesToJSON(value?: EntCustomerEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'buys': value.buys === undefined ? undefined : ((value.buys as Array<any>).map(EntBillToJSON)),
    };
}
