// plugin-welcome/src/api/models/EntProduct.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntProductEdges,
    EntProductEdgesFromJSON,
    EntProductEdgesFromJSONTyped,
    EntProductEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntProduct
 */
export interface EntProduct {
    /**
     * 
     * @type {EntProductEdges}
     * @memberof EntProduct
     */
    edges?: EntProductEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntProduct
     */
    id?: number;
    /**
     * ProductCost holds the value of the "product_cost" field.
     * @type {number}
     * @memberof EntProduct
     */
    productCost?: number;
    /**
     * ProductName holds the value of the "product_name" field.
     * @type {string}
     * @memberof EntProduct
     */
    productName?: string;
}

export function EntProductFromJSON(json: any): EntProduct {
    return EntProductFromJSONTyped(json, false);
}

export function EntProductFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntProduct {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntProductEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'productCost': !exists(json, 'product_cost') ? undefined : json['product_cost'],
        'productName': !exists(json, 'product_name') ? undefined : json['product_name'],
    };
}

export function EntProductToJSON(value?: EntProduct | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntProductEdgesToJSON(value.edges),
        'id': value.id,
        'product_cost': value.productCost,
        'product_name': value.productName,
    };
}
