// plugin-welcome/src/api/models/EntUserEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBill,
    EntBillFromJSON,
    EntBillFromJSONTyped,
    EntBillToJSON,
    EntRole,
    EntRoleFromJSON,
    EntRoleFromJSONTyped,
    EntRoleToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntUserEdges
 */
export interface EntUserEdges {
    /**
     * 
     * @type {EntRole}
     * @memberof EntUserEdges
     */
    role?: EntRole;
    /**
     * Sells holds the value of the sells edge.
     * @type {Array<EntBill>}
     * @memberof EntUserEdges
     */
    sells?: Array<EntBill>;
}

export function EntUserEdgesFromJSON(json: any): EntUserEdges {
    return EntUserEdgesFromJSONTyped(json, false);
}

export function EntUserEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntUserEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'role': !exists(json, 'role') ? undefined : EntRoleFromJSON(json['role']),
        'sells': !exists(json, 'sells') ? undefined : ((json['sells'] as Array<any>).map(EntBillFromJSON)),
    };
}

export function EntUserEdgesToJSON(value?: EntUserEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'role': EntRoleToJSON(value.role),
        'sells': value.sells === undefined ? undefined : ((value.sells as Array<any>).map(EntBillToJSON)),
    };
}
